### ISSUE TITLE
PyPDF2.errors.PageSizeNotDefinedError

### ISSUE THREAD

**Issue by sergueif** [chongchonghe/booklet-creator#1]
Hey I'm getting this error.
What version of python and the packages does this need?

**Comment by chongchonghe** [comment]
As long as you have a working PyPDF2, this program should work. I believe PyPDF2 [requires Python3.6+](https://pypdf2.readthedocs.io/en/latest/meta/CHANGELOG.html).

**Comment by sergueif** [comment]
I tried with Python 3.8.9 and 
```
Package    Version
---------- -------
numpy      1.23.3
pip        22.2.2
PyPDF2     2.10.6
setuptools 63.2.0
```

and get same error
```
Traceback (most recent call last):
  File "/Users/serguei/workspace/booklet-creator/booklet.py", line 70, in <module>
    main(sys.argv[1])
  File "/Users/serguei/workspace/booklet-creator/booklet.py", line 46, in main
    new.addBlankPage()
  File "/Users/serguei/.asdf/installs/python/3.10.7/lib/python3.10/site-packages/PyPDF2/_writer.py", line 371, in addBlankPage
    return self.add_blank_page(width, height)
  File "/Users/serguei/.asdf/installs/python/3.10.7/lib/python3.10/site-packages/PyPDF2/_writer.py", line 358, in add_blank_page
    page = PageObject.create_blank_page(self, width, height)
  File "/Users/serguei/.asdf/installs/python/3.10.7/lib/python3.10/site-packages/PyPDF2/_page.py", line 327, in create_blank_page
    raise PageSizeNotDefinedError
PyPDF2.errors.PageSizeNotDefinedError
```

**Comment by sergueif** [comment]
fwiw, the input pdf had only 1 page in it, which perhaps this code did not expect.

I ran it on a file with 44 pages, but the pages in the output are not in the right order. The last page in input is p68, but in the output, the first page is 46. I thought the output must start with the last page and then alternate 2 pages from front, 2 pages from back, repeat.

**Comment by chongchonghe** [comment]
> fwiw, the input pdf had only 1 page in it, which perhaps this code did not expect.
> 
> I ran it on a file with 44 pages, but the pages in the output are not in the right order. The last page in input is p68, but in the output, the first page is 46. I thought the output must start with the last page and then alternate 2 pages from front, 2 pages from back, repeat.

I fixed the bug that occurred when the number of pages is 1. Just do a pull. You shouldn't use this program on a single-page PDF anyway.

You don't expect the first page of the new doc to be the last page of the old doc.  In fact, the last page of the new doc is the first page of the old doc,  For a four-page document, the reordered page order is [2, 3, 4, 1].

**Comment by sergueif** [comment]
Hey I just saw you made a quick change to fix it. Thank you!

That's interesting. Maybe there's more than one ordering of pages that still results in a valid booklet?
I forked this repo ( https://github.com/sergueif/booklet-creator/commit/64921076a856d19c702c5d5f77e042d2dc9e87cb ) and changed the algorithm to the one described in this page:
https://www.macintoshhowto.com/pages-and-publishing/how-to-print-to-a-pdf-booklet-in-os-x-mojave.html
```
You need to rearrange the pages to be in the right order to make a booklet. This is the order you need:

4 pages: 4,1,2,3.

8 pages: 8,1,2,7,6,3,4,5.

16 pages: 16,1,2,15,14,3,4,13,12,5,6,11,10,7,8,9.

Can you see the pattern?

The pattern is Last, first x2, last x2, first x2 , last x2 etc.

In practice the way to do this is: Grab the Last page then delete it, grab the First page then delete it, grab the new first page then delete it, grab the last Page x2 etc until all the pages are gone.
```

So for 4 pages, your booklet is: [2,3,4,1], but the other one is [4,1,2,3]. The latter is the order I'm used to because the printed output is the same that I used to get with the expensive "Create Booklet" program (https://apps.apple.com/us/app/create-booklet-2/id1350225911?mt=12)

This Github issue can be closed. Much appreciate your super fast responses and hope more people will find this script useful. Take a look at my fork if you're curious. My hunch is that both our repos generate a valid booklet in the end even if the order of pages is different. ( I used my fork on the 44 page PDF I needed a booklet of and it printed perfectly)
